src/waveform/renderers/waveformmark.h

```diff
@@ -1,39 +1,35 @@
 #ifndef WAVEFORMMARK_H
 #define WAVEFORMMARK_H
 
-#include <QString>
+#include <QDomNode>
 #include <QImage>
-#include <QColor>
 
-#include "preferences/usersettings.h"
-#include "skin/skincontext.h"
+#include "util/memory.h"
+
+#include "waveform/renderers/waveformmarkproperties.h"
 
 class ControlProxy;
-class QDomNode;
+class SkinContext;
 class WaveformSignalColors;
 
 class WaveformMark {
   public:
     WaveformMark();
-    ~WaveformMark();
+
     void setup(const QString& group, const QDomNode& node,
                const SkinContext& context,
                const WaveformSignalColors& signalColors);
-    void setKeyAndIndex(const ConfigKey& key, int i);
 
-  private:
-    ControlProxy* m_pPointCos;
+    std::unique_ptr<ControlProxy> m_pPointCos;
 
-    QColor m_color;
-    QColor m_textColor;
-    QString m_text;
-    Qt::Alignment m_align;
-    QString m_pixmapPath;
+    const WaveformMarkProperties& properties() const;
+    void setProperties(const WaveformMarkProperties& properties);
+
+  private:
+    WaveformMarkProperties m_properties;
     QImage m_image;
```

Comment:
That's ok for me. As I mentioned, I didn't take into account all the details.

On 21.08.2016 15:38, Ferran Pujol Camins wrote:
>
> In src/waveform/renderers/waveformmark.h 
> <https://github.com/mixxxdj/mixxx/pull/992#discussion_r75595330>:
>
> >
> > -    QColor m_color;
> > -    QColor m_textColor;
> > -    QString m_text;
> > -    Qt::Alignment m_align;
> > -    QString m_pixmapPath;
> > +    const WaveformMarkProperties& properties() const;
> > +    void setProperties(const WaveformMarkProperties& properties);
> > +
> > +  private:
> > +    WaveformMarkProperties m_properties;
> >      QImage m_image;
>
> Is this worth it? Making m_image part of WaveformMarkProperties means that 
> in WaveformRenderMark::slotCuesUpdated we are copying it just to 
> regenerate it afterwards.
>
> Also, it is strange to me to think about it as a "property" in the same 
> sense than color or text. Conceptually, for me it should be a member of 
> WaveforMark.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/992/files/2f8863b7b8266565a97581fd38cfa6e87ab0ac36#r75595330>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-alocrPGQJL7xYxSVnZdXHpOTqALwks5qiFTvgaJpZM4Jj7vm>.
>